### ISSUE TITLE
ES highlighting: unknown field max_analyzed_offset

### ISSUE THREAD

**Issue by lawliet19189** [elastic/elasticsearch-py#1905]
**Elasticsearch version** (`bin/elasticsearch --version`): 7.10.2

**`elasticsearch-py` version (`elasticsearch.__versionstr__`)**:  7.10.1

**Description of the problem including expected versus actual behavior**: 

+  using [max_analyzed_offset](https://www.elastic.co/guide/en/elasticsearch/reference/current/highlighting.html#highlighting-settings) in search query payload throws the below error

```
RequestError: RequestError(400, 'x_content_parse_exception', '[1:1624] [highlight] unknown field [max_analyzed_offset]')
```

The query highlight payload is the below:
```
{
    'order': 'score',
     'fields': {
        'name': {
            'type': 'plain',
            'number_of_fragments': 1,
            'fragment_size': 80
        },
        'content': {
            'type': 'unified',
            'number_of_fragments': 3,
            'fragment_size': 200,
            'boundary_max_scan': 80
        },
     }
    'max_analyzed_offset': 1000
}
```

I have also tried placing `max_analyzed_offset` both on the root and also inside ['fields']['content'], but nothing seems to work!

Thanks in Advance for your help!

**Comment by sethmlarson** [comment]
Could you include the entire piece of Python code that's failing, otherwise I can't tell where and how you're passing this data to Elasticsearch. Thanks!

**Comment by lawliet19189** [comment]
hey @sethmlarson, I have tried to take a snippet of the full python code and share it below, hopefully, this helps. Please do let me know if you require more information.

```
from elasticsearch import Elasticsearch, RequestsHttpConnection

host, index = ..., ...

es_client = Elasticsearch(
    hosts=[{"host": host, "port": 443}],
    use_ssl=True,
    verify_certs=True,
    connection_class=RequestsHttpConnection,
    retry_on_timeout=True,
)

es_query = {
    "collapse": ...,
    "size": ...,
    "from": ...,
    "track_total_hits": ...,
    "_source": ...,
    "highlight": {
        'order': 'score',
         'fields': {
            'name': {
                'type': 'plain',
                'number_of_fragments': 1,
                'fragment_size': 80
            },
            'content': {
                'type': 'unified',
                'number_of_fragments': 3,
                'fragment_size': 200,
                'boundary_max_scan': 80
            },
         }
    'max_analyzed_offset': 1000
    },
    "query": ...
}

search_response = es.search(body=es_query, index=index)

```
Forgot to mention that we are using opensearch!

**Comment by sethmlarson** [comment]
Ah, that makes sense. The `max_analyzed_offset` field was added in 7.12.0 and OpenSearch is stuck on 7.10.0 APIs. There's not much I can do here unfortunately so will likely have to close. Sorry about that!

Related: if you're interested in using this API and all the others added since 7.10 Elasticsearch and Elastic Cloud are available [through the AWS marketplace](https://aws.amazon.com/marketplace/pp/prodview-voru33wi6xs7k). We just added a free trial option to the AWS marketplace recently which is handy :)
